omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/scripts/dispatch.lua.erb

```diff
@@ -44,9 +44,15 @@ if mode == "api" then
         ngx.exit(ngx.HTTP_OK)
       end
     else
-      -- the request did not originate with a chef client, we'll give something
-      -- more friendly to web browsers.
-      return ngx.exec("/index.html")
+      <% if node['private_chef']['nginx']['show_welcome_page'] %>
+        -- the request did not originate with a chef client, we'll give something
+        -- more friendly to web browsers.
+        return ngx.exec("/index.html")
+      <% else %>
+        -- return 404 because the welcome page has been disabled
+        ngx.status = ngx.HTTP_NOT_FOUND
+        ngx.exit(ngx.HTTP_OK)
```

Comment:
I don't know much about `ngx_lua`, but [these docs](https://github.com/openresty/lua-nginx-module#ngxexit) make me think we might be able to go with a single `ngx.exit(ngx.HTTP_NOT_FOUND)` call. But you've potentially tried that already, if so please ignore `:)`

Reply:
@srenatus Yeah, I did try that. When you use `ngx.exit(ngx.HTTP_NOT_FOUND)` only you get back the below page
![screen shot 2017-11-29 at 4 33 53 pm](https://user-images.githubusercontent.com/1679247/33406754-3d58da9e-d523-11e7-827a-5a6700b8f474.png)

If you leave it the way it is you get back nothing in the response body with the status of 404. I think this would be better to not return anything in the response body. This would better hide the fact that this is a chef-server endpoint.